spikes/ORY_KRATOS.md

```diff
@@ -0,0 +1,84 @@
+
+Ory Kratos is a fairly new and interesting project that is trying to solve the problem of authentication and authorization in a very modern way. It is a zero trust identity and access management (IAM) solution that can be used to secure web applications and APIs. It is a cloud native solution.
+
+Languages and tools they say Ory Kratos works with are Swift, React.js, Angular (client side), PHP, Ruby, Python, Java (server side), Node.js, and TypeScript. Database backends supported are SQLite, PostgreSQL, MySQL, and CockroachDB.
+
+### Implementation
+
+The QG web application will use Kratos to provide
+
+-   user registration
+-   user login
+-   user logout
+-   user profile management
+-   user password management
+-   user email management
+-   user 2FA management
+-   social sign ins
+
+To try Kratos out by following [the Quick Start tutorial and example app](https://www.ory.sh/docs/kratos/quickstart) you need to know and implement Docker, Docker Compose, Git, Node JS, and Typescript. The end goal will be to create a simple app entry flow for logging in,logging out, and registration. The quick start exercise also uses MailSlurper, a mock SMTP server for testing email verification.
+
+You have to keep an eye on ports 4455, 4433, 4434, and 4436 to make sure they are free. If they are not free, you can change the ports in the docker-compose.yml file. Or in the command line of a Mac, you can view the ports with `lsof -i :<port number>` and kill the process with `kill -9 <PID>`. On a Windows machine, you can view the ports with `netstat -ano | findstr :<port number>` and kill the process with `taskkill /PID <PID> /F`.
+
+Another thing: Use `127.0.0.1` and NOT `localhost`. TIL from the tut that browsers treat these two as separate domains and will therefore have issues with setting and using cookies!
+
+## ✅ Pros
+
+- Kratos gives the options of HTTP redirects and anti-CSRF tokens
+- It is a standalone service that can be used with any programming language and framework. If you want to change the backend framework, you can easier than expected. Can be self-hosted or otherwise. This makes it low coupling.
+- An app doesn't hav to store PII for the user's identity or for the sign in/sign up flow.
+
+## 🛑 Cons
+
+- Kratos uses cookies in order for the anti-CSRF tokens to work
+- Kratos is absolutely NOT beginner-friendly! Beginners tend to not be familiar with cloud computing, session tokens, Docker or Kubernetes. Kratos is a cloud native solution that can be deployed on tools like this.
+
+## ⌛ Timebox
+
+## 👀 Review
+
+## 🔚 Conclusion
+I don't believe that this tool should be used in the QG app UNLESS there is a proper demonstration from the original suggestor, to explain why the tool is appropriate for the QG app, how to implement it, and how to administer the features of it for long-term support. Even though it is, supposedly, low maintenance, it is not low maintenance for a beginner. It is not beginner-friendly. 
```

Comment:
Let's start a conversation about alternatives in the development chat.

Reply:
This is a really good point, in that we need something that can be administered by volunteers who won't always have experience with it.